lib/active_record_query_trace.rb

```diff
@@ -99,7 +100,8 @@ def fully_formatted_trace
 
     # Must be called after the backtrace cleaner.
     def lines_to_display(full_trace)
-      ActiveRecordQueryTrace.lines.zero? ? full_trace : full_trace.first(ActiveRecordQueryTrace.lines)
+      lines = ActiveRecordQueryTrace.lines || (ActiveRecordQueryTrace.level == :full ? 0 : DEFAULT_LINES)
```

Comment:
The log level (`:full`, `:app` or `:rails`) setting is not meant to affect the number of lines that are displayed in the backtrace. `:full` means that it should display lines related to both Rails and application files. The `lines` and `level` should remain separate.